src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsActionClass.kt

```diff
@@ -18,32 +20,42 @@ class GenerateTestsActionClass : AnAction() {
     /**
      * Performs test generation for a class when the action is invoked.
      *
-     * @param e AnActionEvent class that contains useful information about the action event
+     * @param e an action event that contains useful information
      */
     override fun actionPerformed(e: AnActionEvent) {
-        // determine class path
         val project: Project = e.project ?: return
 
+        val psiFile: PsiFile = e.dataContext.getData(CommonDataKeys.PSI_FILE) ?: return
+        val caret: Caret = e.dataContext.getData(CommonDataKeys.CARET)?.caretModel?.primaryCaret ?: return
+
+        val psiClass: PsiClass = GenerateTestsUtils.getSurroundingClass(psiFile, caret) ?: return
+        val classFQN = psiClass.qualifiedName ?: return
+
         val projectPath: String = ProjectRootManager.getInstance(project).contentRoots.first().path
         val projectClassPath = "$projectPath/target/classes/"
 
         log.info("Generating tests for project $projectPath with classpath $projectClassPath")
 
-        val psiClass = e.dataContext.getData(CommonDataKeys.PSI_ELEMENT) as PsiClass // Checked in update method
```

Comment:
why did you move this several lines up?

Reply:
I think that i was changing a lot of stuff in this method and in the end ended up with that order. But i like it more